We're going to make it clearer to users that they should NOT edit
  this file (as installed to /usr/share/hal/fdi/... and also let
  them know where to copy it and how to edit that copy for an
  alternate keymap and such...   --rworkman @ 20091211

diff -Nur hal-0.5.11.orig/fdi/policy/10osvendor/10-keymap.fdi hal-0.5.11/fdi/policy/10osvendor/10-keymap.fdi
--- hal-0.5.11.orig/fdi/policy/10osvendor/10-keymap.fdi	2008-05-07 18:21:16.000000000 -0500
+++ hal-0.5.11/fdi/policy/10osvendor/10-keymap.fdi	2009-07-21 00:06:11.779152226 -0500
@@ -1,4 +1,8 @@
 <?xml version="1.0" encoding="ISO-8859-1"?> <!-- -*- SGML -*- -->
+
+<!-- DO NOT EDIT THIS FILE - IT WILL BE OVERWRITTEN ON UPGRADES.
+     SEE THE "READ THIS" NOTE BELOW FOR INSTRUCTIONS -->
+
 <deviceinfo version="0.2">
   <device>
     <match key="info.capabilities" contains="input.keymap">
@@ -6,18 +10,41 @@
     </match>
 
     <match key="info.capabilities" contains="input.keys">
-      <merge key="input.xkb.rules" type="string">base</merge>
 
-      <!-- If we're using Linux, we use evdev by default (falling back to
-           keyboard otherwise). -->
-      <merge key="input.xkb.model" type="string">keyboard</merge>
-      <match key="/org/freedesktop/Hal/devices/computer:system.kernel.name"
-             string="Linux">
-        <merge key="input.xkb.model" type="string">evdev</merge>
-      </match>
+      <!-- Restore Ctrl-Alt-Bksp Xserver Zapping -->
+      <merge key="input.xkb.options" type="string">terminate:ctrl_alt_bksp</merge>
 
+      <!-- Edit (as needed) these four lines in the copied fdi file -->
+      <merge key="input.xkb.rules" type="string">base</merge>
+      <merge key="input.xkb.model" type="string">evdev</merge>
       <merge key="input.xkb.layout" type="string">us</merge>
       <merge key="input.xkb.variant" type="string" />
+
     </match>
   </device>
 </deviceinfo>
+
+<!-- READ THIS FOR CUSTOM KEYBOARD INFORMATION
+
+  If you want to add a custom model/layout/variant to X, you will need to COPY
+  this file to /etc/hal/fdi/policy/ and edit that copy.  After editing it to
+  suit, you will need to leave X, restart the HAL daemon, and start X again.
+
+  Here's an example of the four lines from above:
+      <merge key="input.xkb.rules" type="string">base</merge>
+      <merge key="input.xkb.model" type="string">evdev</merge>
+      <merge key="input.xkb.layout" type="string">us</merge>
+      <merge key="input.xkb.variant" type="string">intl</merge>
+
+  Many desktop environments, including KDE and Xfce, have their own methods to
+  configure keyboard layouts and such if you'd like to use them.
+
+  If you prefer to use the "old" way of configuring keyboards (without HAL's
+  input device hotplugging), then you'll need to add the following lines to
+  the ServerFlags section of /etc/X11/xorg.conf:
+    Option   "AllowEmptyInput"     "false"
+    Option   "AutoAddDevices"      "false"
+    Option   "AutoEnableDevices"   "false"
+  You can now edit the Keyboard section of xorg.conf as usual.
+-->
+
